gemrb/plugins/FXOpcodes/FXOpcodes.cpp

```diff
@@ -2367,7 +2367,29 @@ int fx_to_hit_modifier (Scriptable* /*Owner*/, Actor* target, Effect* fx)
 {
 	if(0) print("fx_to_hit_modifier(%2d): Mod: %d, Type: %d", fx->Opcode, fx->Parameter1, fx->Parameter2);
 
-	target->ToHit.HandleFxBonus(fx->Parameter1, fx->TimingMode==FX_DURATION_INSTANT_PERMANENT);
+	int percentage;
+	switch (fx->Parameter2) {
+	case MOD_ADDITIVE:
+	default:
+		target->ToHit.HandleFxBonus(fx->Parameter1, fx->TimingMode==FX_DURATION_INSTANT_PERMANENT);
+		break;
+	case MOD_ABSOLUTE:
+		if (fx->TimingMode == FX_DURATION_INSTANT_PERMANENT) {
+			target->ToHit.SetBase(fx->Parameter1);
+		} else {
+			//FIXME: two such effects probably should not stack, but they do for now
+			target->ToHit.SetFxBonus(fx->Parameter1 - target->ToHit.GetBase(), MOD_ADDITIVE);
```

Comment:
This is tricky. The generic bonus is also based on effects, so undoing things properly is harder. The whole system was added for iwd2, while bg2 just used one main stat. According to IESDP, it only worked on the base stat too, but there could have been a mistake.

So in this case we'd ideally be affecting the total. Not sure if it's worth bothering with, since I don't know of any users.

The more actionable problem is that this line doesn't work for iwd2 — it would just set the bonus to param1, since bab starts at 0.

Reply:
In the BG2 section IESDP says the opcode works on modified THAC0, but IIRC it turned out not to be true for potions of heroism (which were the reason for this fix).  I'll recheck this.
IDW2 may need special handing anyway, since from what I understand changes to BAB can also affect APR.  Or not?
Also I don't get why `ToHit.GetBase()` would return 0 for IWD2?  `Actor::SetBaseAPRandAB()` seems to set base pretty much at the same time both for 2e and 3e.

Reply:
Okay, so I tested a bit on my BG2 install and the percentage modifier, at least, does not respect previous changes by the additive modifiers of the same opcode, just as I remembered.  I'm pretty sure by now that it does operate on base THAC0 (but I haven't tested _all_ cases).  The absolute modifier is indeed non-cumulative with itself; furthermore, it's not cumulative with the percentage modifiers either!  (Only the latest applied works.)  The percentage modifier is not cumulative with itself, either.  It seems that the percentage type works more like absolute type behind the scenes, rather than like a bonus.  The most faithful to original way to reproduce it would be, it seems, to create _yet another_ ToHit field that would be populated exclusively by those two modifier types.  Or we could make `base` rewritable without affecting `Owner`.
On the topic of users: after searching with DLTC Editor, it seems that at least some "natural weapon" items do use the absolute modifier of this opcode, e.g. `SLAYERWP.ITM`.  They're by nature unstackable, though.

Reply:
IWD2: I was just going off the fact that base starts at 0, not 20, but didn't check how we store it. 

Likely not worthy to lose more time on this. What some people could miss is the tobex extension, which is anothermod type though. It was introduced to handle stacking better.